### PR TITLE
FeatureTable udf support multi column

### PR DESCRIPTION
**Pull request by hkvision**

### CODE REVIEW

File: pyzoo/zoo/models/recommendation/wide_and_deep.py [hunk 1]
```diff
@@ -90,6 +90,15 @@ def __str__(self):
...
 
+    @property
+    def feature_cols(self):
+        return self.wide_base_cols + self.wide_cross_cols +\
+            self.indicator_cols + self.embed_cols + self.continuous_cols
+
+    @property
+    def label_cols(self):
+        return [self.label]
+
```
[the remark this answers is not in the document]
Comment (hkvision) [reply pyzoo/zoo/models/recommendation/wide_and_deep.py:101]: But we can reuse ColumnFeatureInfo

File: pyzoo/zoo/friesian/feature/table.py [hunk 2]
```diff
@@ -1168,15 +1168,23 @@ def apply(self, in_col, out_col, func, dtype="string"):
         """
         Transform a FeatureTable using a user-defined Python function.
 
-        :param in_col: str, the name of column to be transformed.
+        :param in_col: str or a list of str, the column(s) to be transformed.
```
[the remark this answers is not in the document]
Comment (hkvision) [reply pyzoo/zoo/friesian/feature/table.py:1171]: The columns can have different types, no?

File: pyzoo/zoo/models/recommendation/wide_and_deep.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jason-dai) [reply pyzoo/zoo/models/recommendation/wide_and_deep.py:101]: > But we can reuse ColumnFeatureInfo

Does it makes sense to define a custom class? Will it be easier for the users to use something like dictionary? How's that handled in tensorflow?

File: pyzoo/zoo/friesian/feature/table.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hkvision) [reply pyzoo/zoo/friesian/feature/table.py:1171]:
```
transform = lambda x: 1
train_tbl = train_tbl.apply(["reply_label", "len_hashtags", "text_tokens"], "o1", transform, dtype="int")
```
The above code works for different columns actually.

File: pyzoo/zoo/models/recommendation/wide_and_deep.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkvision) [reply pyzoo/zoo/models/recommendation/wide_and_deep.py:101]: I think for our current implementation we can just use list or dict...

https://github.com/analytics-zoo/friesian/blob/recsys-challenge/wechat2021/baseline/wideANDdeep/baseline.py#L181
TF uses Column classes to combine column name and dimensions.

**Comment by hkvision** [comment]
jenkins: http://10.112.231.51:18888/job/ZOO-PR-Validation/6266/
The failures are irrelevant.

File: pyzoo/zoo/models/recommendation/wide_and_deep.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenniew) [reply pyzoo/zoo/models/recommendation/wide_and_deep.py:101]: Yes, we may create similar API as tf to wrap different columns.

**Pull request merged**